File: Bookworm Backend Dot net/Models/Genre.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace Bookworm.Models
{
    [Table("genre_master")]
    public class Genre
    {
        [Key]
        [Column("genre_id")]
        public int Id { get; set; }

        [Required]
        [StringLength(25)]
        [Column("genre_desc")]
        public string Description { get; set; }
    }
}

File: Bookworm Backend Spring Boot/src/main/java/com/bookworm/entities/Genre.java
package com.bookworm.entities;


import jakarta.persistence.*;
import lombok.*;
;

/**
 * Represents a genre for categorizing products.
 * Maps to the 'genres' table.
 */
@Entity
@Table(name = "genre_master")
@Getter
@Setter
public class Genre {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "genre_id")
    private Integer id;

    @Column(name = "genre_desc", length = 25, nullable = false)
    private String description;

    
}